privacyidea/lib/auditmodules/loggeraudit.py

```diff
@@ -71,6 +71,8 @@ class Audit(AuditBase):
 
     def __init__(self, config=None):
         super(Audit, self).__init__(config)
+        # overwrite the startdate from the superclass with the isoformat
+        self.audit_data["startdate"] = datetime.utcnow().isoformat()
```

Comment:
why don't You use `isoformat()` in the base class as well?

And shouldn't SQLALchemy be able to handle datetime objects as well?

Reply:
There was a reason, I can not remeber. Need to check this.

Reply:
OK, here is the ting!
The SQLAudit actually uses the baseclass, where "startedate" is set to datetime object using

      datetime.datetime.now()

So just like the "enddate" the "startdate" is stored as a datetime object.

See the old "enddate"

https://github.com/privacyidea/privacyidea/blob/master/privacyidea/models.py#L2627

See the startdate in the baseclass.

The isoformat is simply used here, since it is the loggeraudit, which actually writes a *string* to a file. So we can not store a datetime object.

Reply:
We could simply add a json decoder for datetime: https://pynative.com/python-serialize-datetime-into-json/
That way we don't have to take extra care for the logger audit module.

Reply:
The logger currently also writes the end date in isoformat. This is why I simply wanted to keep it that way like it currently works.
https://github.com/privacyidea/privacyidea/blob/master/privacyidea/lib/auditmodules/loggeraudit.py#L84
I do not see any reason to change the functionality of the logger audit here. 
And I do not see any reason why one date in the loggeraudit should be isoformat and the other one json.

Reply:
i am ok with this.